### PR TITLE
PP-12915: Request stripe test account endpoint

### PR DESCRIPTION
**Pull request by oswaldquek**
Wire up classes for the `/v1/service/{serivceId}/request-stripe-test-account` endpoint. An initial test to ensure we've the correct Stripe endpoints are called.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java [hunk 1]
```diff
@@ -35,6 +39,27 @@ public StripeAccountResource(StripeAccountService stripeAccountService,
         this.stripeAccountService = stripeAccountService;
         this.gatewayAccountService = gatewayAccountService;
     }
+    
+    @POST
+    @Path("/v1/service/{serviceId}/request-stripe-test-account")
+    // TODO uncomment when this endpoint's functionality is complete
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java:46]: is this endpoint not finished yet?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 2]
```diff
@@ -34,10 +36,10 @@ public class StripeAccountService {
     
     private static RequestOptions requestOptions;
     
-//    @Inject
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java:37]: can this commented out code be removed?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java [hunk 3]
```diff
@@ -35,6 +39,27 @@ public StripeAccountResource(StripeAccountService stripeAccountService,
...
+//            responses = {
+//                    @ApiResponse(responseCode = "200", description = "OK"),
+//                    @ApiResponse(responseCode = "400", description = "Stripe Connect Account already exists"),
+//                    @ApiResponse(responseCode = "404", description = "Not found - Account with serviceId does not exist")
+//            }
+//    )
+    public Response requestStripeTestAccount(
+            @Parameter(example = "46eb1b601348499196c99de90482ee68", description = "Service ID") @PathParam("serviceId") String serviceId
+    ) {
+        String serviceName = gatewayAccountService.getGatewayAccountByServiceIdAndAccountType(serviceId, TEST).get().getServiceName();
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: no check here to verify optional value is present before accessing

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java [hunk 4]
```diff
@@ -34,10 +36,10 @@ public class StripeAccountService {
...
-//    public StripeAccountService(StripeGatewayConfig stripeGatewayConfig) {
-//        requestOptions = RequestOptions.builder().setApiKey(stripeGatewayConfig.getAuthTokens().getTest()).build();
-//    }
+    @Inject
+    public StripeAccountService(StripeGatewayConfig stripeGatewayConfig) {
+        requestOptions = RequestOptions.builder().setApiKey(stripeGatewayConfig.getAuthTokens().getTest()).build();
```
Comment (nlsteers) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java:41]: imo static fields shouldn't be updated in constructors, does `requestOptions` need to be static?

File: src/test/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountServiceTest.java [hunk 5]
```diff
@@ -30,10 +37,18 @@ class StripeAccountServiceTest {
     private StripeAccountService stripeAccountService;
 
     private GatewayAccountEntity gatewayAccountEntity;
+    
+    @Mock
+    private StripeGatewayConfig stripeGatewayConfig;
+    
+    @Mock
+    private StripeAuthTokens stripAuthTokens;
```
Comment (nlsteers) [inline src/test/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountServiceTest.java]: field name typo

**Review by nlsteers — COMMENTED** [review]
couple of comments to address, good work

File: src/test/resources/templates/stripe/create_external_account_response.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
+  "id": "ba_1NAiJy2eZvKYlo2CvChQKz5k",
+  "object": "bank_account",
+  "account": "acct_1032D82eZvKYlo2C",
+  "account_holder_name": "Jane Austen",
+  "account_holder_type": "company",
+  "account_type": null,
+  "bank_name": "STRIPE TEST BANK",
+  "country": "US",
```
Comment (nlsteers) [inline src/test/resources/templates/stripe/create_external_account_response.json]: can we use `GB` and `gbp` please

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 2 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java:37]: But it is removed in this PR...

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java (hunk 1 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java:46]: Yeah it's not as I have [some remaining functionality to implement](https://github.com/alphagov/pay-connector/pull/5461/files#diff-7080f046a6941a42f2da4dadcc93a304f14334e5603e9a5d68277a28e581b261R48-R50)

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java (hunk 3 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: Yeah I will verify this when I've implemented [this test](https://github.com/alphagov/pay-connector/pull/5461/files#diff-7080f046a6941a42f2da4dadcc93a304f14334e5603e9a5d68277a28e581b261R67-R70). Can we leave this here for now?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 2 above)
Comment (nlsteers) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java:37]: oh yeah lol

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java (hunk 3 above)
Comment (nlsteers) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: easier to miss in review if not fixed when added

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java (hunk 4 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountService.java:41]: fixed

File: src/test/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountServiceTest.java (hunk 5 above)
Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccount/service/StripeAccountServiceTest.java]: fixed

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java (hunk 3 above)
Comment (nlsteers) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: i'll make a note for the next PR :D

File: src/test/resources/templates/stripe/create_external_account_response.json (hunk 6 above)
Comment (oswaldquek) [reply src/test/resources/templates/stripe/create_external_account_response.json]: fixed

File: src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java (hunk 3 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/resource/StripeAccountResource.java]: I've dealt with this by implementing the [service not found test](https://github.com/alphagov/pay-connector/pull/5461/commits/9309ea6df64aaaeb083458028964a0b2935190e9) 🙂

**Review by nlsteers — APPROVED** [review]
LGTM

**Pull request merged**